### PR TITLE
Automatically load commands

### PR DESCRIPTION
**Pull request by laurelmay**
The bot help information (which isn't really a command) moves out of the commands package. And the update Cog (which also wasn't a command) moves out as well. This should make it quite a bit easier to add new commands going forward.

I'm not sure what the testing procedure is.

### CODE REVIEW

**Comment by ajsnarr98** [comment]
This is really cool.

I can test it for you. Although testing the update checker I cannot do. Right now there are no unit tests. I wonder if there is a way we could use mocks to make unit tests for the discord api.

File: lazybot/commands/__init__.py [hunk 1]
```diff
@@ -7,8 +7,88 @@
...
+        }
+    )
+
+    for module in modules:
+        path = os.path.join(directory, module)
+        modname = os.path.splitext(os.path.basename(path))[0]
+
+        try:
+            loader = SourceFileLoader(f"{__name__}.{modname}", path)
+            command_module = type.ModuleType(loader.name)
```
Comment (ajsnarr98) [inline lazybot/commands/__init__.py]: should be `types.ModuleType(loader.name)`

File: lazybot/bot.py [hunk 2]
```diff
@@ -120,11 +115,10 @@ async def on_member_join(member):
 ### Commands ###
 ################
 
-bot.add_cog(CommandColor(bot))
-bot.add_cog(CommandGradYear(bot))
-bot.add_cog(MiscFun(bot))
+for command in all_commands:
```
Comment (ajsnarr98) [inline lazybot/bot.py]: `for command in all_commands():`

File: lazybot/commands/__init__.py [hunk 3]
```diff
@@ -7,8 +7,88 @@
...
+                getattr(command_module, content_name)
+                for content_name in dir(command_module)
+            ]
+            _COMMANDS.append(
+                {
+                    command_class
+                    for command_class in module_contents
+                    if _is_loaded_subclass(command_class, Cog)
+                }
+            )
```
Comment (ajsnarr98) [inline lazybot/commands/__init__.py]: `_COMMANDS.extend(list(
                {
                    command_class
                    for command_class in module_contents
                    if _is_loaded_subclass(command_class, Cog)
                }
            ))`

**Comment by laurelmay** [comment]
I validated that all previous commands are still detected and run correctly. Additionally, I validated update detection and autoupdating by pushing an additional commit to this branch that I didn't have locally and the bot detected the update and was able to update successfully.

**Comment by ajsnarr98** [comment]
Pull request approved to merge ✅

**Pull request merged**